cli/azd/pkg/exec/az_emulator.go

```diff
@@ -0,0 +1,61 @@
+package exec
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+)
+
+const (
+	emulatorEnvName string = "AZURE_AZ_EMULATOR"
+)
+
+// IsAzEmulator returns true if the AZURE_AZ_EMULATOR environment variable is defined.
+// It does not matter the value of the environment variable, as long as it is defined.
+func IsAzEmulator() bool {
+	_, emulateEnvVarDefined := os.LookupEnv(emulatorEnvName)
+	return emulateEnvVarDefined
+}
+
+// creates a copy of azd binary and renames it to az and returns the path to it
+func emulateAzFromPath() (string, error) {
+	path, err := exec.LookPath("azd")
+	if err != nil {
+		return "", fmt.Errorf("azd binary not found in PATH: %w", err)
+	}
+	azdConfigPath, err := config.GetUserConfigDir()
+	if err != nil {
+		return "", fmt.Errorf("could not get user config dir: %w", err)
+	}
+	emuPath := filepath.Join(azdConfigPath, "bin", "azEmulate")
+	err = os.MkdirAll(emuPath, osutil.PermissionDirectoryOwnerOnly)
+	if err != nil {
+		return "", fmt.Errorf("could not create directory for azEmulate: %w", err)
+	}
+	emuPath = filepath.Join(emuPath, strings.ReplaceAll(filepath.Base(path), "azd", "az"))
+
+	srcFile, err := os.Open(path)
+	if err != nil {
+		return "", fmt.Errorf("opening src: %w", err)
+	}
+	defer srcFile.Close()
+
+	destFile, err := os.OpenFile(emuPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+	if err != nil {
+		return "", fmt.Errorf("creating dest: %w", err)
+	}
+	defer destFile.Close()
+
+	_, err = io.Copy(destFile, srcFile)
```

Comment:
Wondering how this may impact concurrent azd operations running on the same machine (especially on Windows). I wonder if we should think about symlinking the current binary, and also create a unique temp directory.

Reply:
creating a unique temp per run makes total sense.  It's hard to think about running 2 azd deployments at the same time form the same machine and have them crash exactly the same time during the binary copy!!!  but still a chance.

I would avoid symlinking, it tends to lead to other issues.  Creating a unique folder per execution should be fine

Reply:
Updated to create unique folder per run